jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -318,9 +397,170 @@ def vue_update_subset(self, *args):
+                            d_att["orig"] = new_orig_xy[0]
+                        if d_att["att"] == 'yc':
+                            d_att["value"] = new_xy[1]
+                            d_att["orig"] = new_orig_xy[1]
+                        if d_att["att"] == 'radius':
+                            d_att["value"] = new_rad
+                            d_att["orig"] = new_orig_rad
+                    if self.subset_types[index] == "EllipticalROI":
+                        if d_att["att"] == 'xc':
+                            d_att["value"] = new_xy[0]
+                            d_att["orig"] = new_orig_xy[0]
+                        if d_att["att"] == 'yc':
+                            d_att["value"] = new_xy[1]
+                            d_att["orig"] = new_orig_xy[1]
+                        if d_att["att"] == 'radius_x':
+                            d_att["value"] = new_rads[0]
+                            d_att["orig"] = new_orig_rads[0]
+                        if d_att["att"] == 'radius_y':
+                            d_att["value"] = new_rads[1]
+                            d_att["orig"] = new_orig_rads[1]
+                        if d_att["att"] == 'theta':
+                            d_att["value"] = new_angle
+                            d_att["orig"] = new_orig_angle
+                    if self.subset_types[index] == "CircularAnnulusROI":
+                        if d_att["att"] == 'xc':
+                            d_att["value"] = new_xy[0]
+                            d_att["orig"] = new_orig_xy[0]
+                        if d_att["att"] == 'yc':
+                            d_att["value"] = new_xy[1]
+                            d_att["orig"] = new_orig_xy[1]
+                        if d_att["att"] == 'inner_radius':
+                            d_att["value"] = new_rads[0]
+                            d_att["orig"] = new_orig_rads[0]
+                        if d_att["att"] == 'outer_radius':
+                            d_att["value"] = new_rads[1]
+                            d_att["orig"] = new_orig_rads[1]
+                    if self.subset_types[index] == "RectangularROI":
+                        if d_att["att"] == 'xmin':
+                            d_att["value"] = new_xmin
+                            d_att["orig"] = new_orig_xmin
+                        if d_att["att"] == 'xmax':
+                            d_att["value"] = new_xmax
+                            d_att["orig"] = new_orig_xmax
+                        if d_att["att"] == 'ymin':
+                            d_att["value"] = new_ymin
+                            d_att["orig"] = new_orig_ymin
+                        if d_att["att"] == 'ymax':
+                            d_att["value"] = new_ymax
+                            d_att["orig"] = new_orig_ymax
+                        if d_att["att"] == 'theta':
+                            d_att["value"] = new_angle
+                            d_att["orig"] = new_orig_angle
```

Comment:
is there any way to handle this just by looping over a dict?  Perhaps `new_*` needs to be stored in a dictionary instead of separate values and with the same keys as `d_att` so this can all just be replaced with something like `d_att = {**d_att, **new_att}`?

Reply:
yep, i will clean this up. i had it listed out individually so i could track down errors but it doesn't need to be this many lines.